dbos/dbos.go

```diff
@@ -372,62 +373,91 @@ func (c *dbosContext) Launch() error {
 	return nil
 }
 
-// Cancel gracefully shuts down the DBOS runtime by canceling the context, waiting for
-// all workflows to complete, and cleaning up system resources including the database
-// connection pool, queue runner, workflow scheduler, and admin server.
+// Cancel cancels the context, waits for workflows to complete, and stops the queue runner and workflow scheduler.
+// This is the core cancellation functionality that signals workflows should stop processing.
 // All workflows and steps contexts will be canceled, which one can check using their context's Done() method.
-//
-// This method blocks until all workflows finish and all resources are properly cleaned up.
-// It should be called when the application is shutting down to ensure data consistency.
-func (c *dbosContext) Cancel() {
-	c.logger.Info("Shutting down DBOS context")
+// Cancel is a permanent operation and should be called when the application is shutting down or deactivated.
+func (c *dbosContext) Cancel(timeout time.Duration) {
+	c.logger.Info("Cancelling DBOS context")
 
 	// Cancel the context to signal all resources to stop
-	c.ctxCancelFunc(errors.New("DBOS shutdown initiated"))
+	c.ctxCancelFunc(errors.New("DBOS cancellation initiated"))
 
 	// Wait for all workflows to finish
 	c.logger.Info("Waiting for all workflows to finish")
-	c.workflowsWg.Wait()
-	c.logger.Info("All workflows completed")
+	done := make(chan struct{})
```

Comment:
Why do we wait for all workflows to finish?

Reply:
It is not mandatory (there is a timeout) but very useful for testing